tools/ARIAtools/product.py

```diff
@@ -714,6 +722,49 @@ def __mappingVersion__(self, fname, version):
                     if '/science/grids/corrections/external/troposphere/' in i:
                         model_name.append(i.split('/')[-3])
 
+                # exit if user does not specify a valid tropo model name
+                if self.tropo_models is not None:
+                    if self.tropo_models.lower() != 'all':
```

Comment:
Sorry, I found an issue with this line when running the regression tests. I found that `self.tropo_models` may be a list, in which case the `self.tropo_models.lower()` raises `AttributeError`. Wrapping this block in
```
                    with contextlib.suppress(AttributeError):
```
avoids that issue, or a similar solution will work

Reply:
Hello @alexfore , I went ahead and adjusted the logic. Before my tweaks, the user input for `tropo_model` was tested for every file in a loop. This is only necessary to check once, and for only Sentinel-1 products (it is not currently relevant for NISAR products). Please let me know what you think. Thanks

Reply:
OK, it passes the tests now